src/cr/cube/matrix/assembler.py

```diff
@@ -1161,6 +1159,43 @@ def _subtotal_values(self):
         return intersections[:, self._insertion_idx]
 
 
+class _SortRowsByLabelHelper(_RowOrderHelper):
+    """Orders elements by the values of their labels (from the dimension)."""
+
+    @lazyproperty
+    def _element_values(self):
+        """Sequence of body labels that form the basis for sort order.
+
+        There is one value per row and values appear in payload (dimension) element
+        order. These are only the "base" values and do not include insertions.
+        """
+        return np.array(self._dimensions[0].element_labels)
+
+    @lazyproperty
+    def _order(self):
+        """tuple of int element-idx specifying ordering of dimension elements."""
+        try:
+            return SortByValueCollator.display_order(
+                self._rows_dimension,
+                self._element_values,
+                self._subtotal_values,
+                self._empty_row_idxs,
+            )
+        except ValueError:
+            return PayloadOrderCollator.display_order(
+                self._rows_dimension, self._empty_row_idxs
+            )
```

Comment:
It's interesting that this exact implementation of `._order` is duplicated in all 5 sort-by-value Helper classes but can't be extracted to the base class `_RowOrderHelper` because that already has one (this implementation is specific to sort-by-value).

The current object hierarchy looks like this:

* `_BaseOrderHelper`
  * `_ColumnOrderHelper`
  * `_RowOrderHelper`
    * `_SortRowsByBaseColumnHelper`
    * `_SortRowsByInsertedColumnHelper`
    * `_SortRowsByLabelsHelper`
    * `_SortRowsByMarginalHelper`

The object hierarchy is already a bit deep, but I'm thinking we should probably extract a `_BaseSortRowsByValueHelper` to be the superclass to the five sort-by-value classes and move this `._order` implementation there.

Reply:
Oh, didn't see you proposed a name. I chose `_SortRowsBaseHelper` to try to cheat and keep things in alphabetical order and also valid.